### ISSUE TITLE
Spyder autocomplete box persists even with all options turned off

### ISSUE THREAD

**Issue by btglau** [spyder-ide/spyder#15088]
## Problem Description
Spyder keeps trying to autocomplete various things that I type:

<img width="383" alt="Screen Shot 2021-04-02 at 3 41 18 PM" src="https://user-images.githubusercontent.com/14989045/113448456-df5c2680-93c9-11eb-9b09-a68a73e7a66a.png">

I've turned off nearly all autocomplete options (except linting) but this problem persists:

<img width="759" alt="Screen Shot 2021-04-02 at 3 45 26 PM" src="https://user-images.githubusercontent.com/14989045/113448765-732df280-93ca-11eb-9268-e8c34bd21f03.png">
<img width="759" alt="Screen Shot 2021-04-02 at 3 45 42 PM" src="https://user-images.githubusercontent.com/14989045/113448782-7c1ec400-93ca-11eb-8eb7-da1b3a735d25.png">

### What steps reproduce the problem?

1. Spyder will highlight existing variables, such as "cro_energy" in this case
2. When you attempt to modify or add to the FIRST CHARACTER of the variable name, spyder then suggests autocomplete even if all options have been turned off
3. Alternatively, add a character to the beginning of the highlighted variable, then delete it. The autocomplete box will show.
4. The issue also occurs when modifying a highlighted variable after a dot.

### What is the expected output? What do you see instead?

No autocomplete box.

## Versions
* Spyder version: 4.2.5 None
* Python version: 3.9.2 64-bit
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Darwin 20.3.0

### Dependencies

# Mandatory:
applaunchservices >=0.1.7       :  0.2.1 (OK)
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.21.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  23.0.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.1 (OK)
pylint >=1.0                    :  2.7.2 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8;<3.0           :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.3               :  5.0.3 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  53.0.0 (OK)
sphinx >=0.6.6                  :  3.5.3 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  22.0.3 (OK)

# Optional:
cython >=0.21                   :  0.29.22 (OK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.20.1 (OK)
pandas >=1.1.1                  :  1.2.3 (OK)
scipy >=0.17.0                  :  1.6.1 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)

**Comment by ccordoba12** [comment]
Hey @btglau, thanks for reporting. This is now fixed in Spyder 5 because you can completely turn off all code completion and linting features (which is not really possible in Spyder 4):

![imagen](https://user-images.githubusercontent.com/365293/113490712-8d7ed380-9491-11eb-80df-65dc131c1711.png)

You can update to this version by downloading our installer here:

https://github.com/spyder-ide/spyder/releases/tag/v5.0.0

Please try that and let us know of it works for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.

**Comment by btglau** [comment]
I'm sorry about the delay in responding. I installed 5.0.0 and the problem is gone, although I did not need to entirely disable code completion and linting. In fact, the only option that worked for me was disabling the Text Snippets Provider (Preferences -> Completion and Linting -> General -> Providers), as nothing else changed the behaviour with respect to this problem (I tried disabling options in Completion and Linting -> Introspection one by one).
